main.go

```diff
@@ -120,22 +120,36 @@ func main() {
 	logger.Printf("Args: %v\n", os.Args)
 	defer logger.Printf("=== Exiting ===\n")
 
+
+	checkUpdateChan := make(chan interface{})
+	go checkUpdateAvailable(checkUpdateChan)
+
 	err := app.Run(os.Args)
 	if err != nil {
 		fmt.Println(err)
 		logger.Println(err)
 	}
 
+	updateAvailable := (<-checkUpdateChan).(bool)
+	if updateAvailable {
+		latestVersion := (<-checkUpdateChan).(string)
+		fmt.Printf("A new version of Edward is available (%v), update with:\n\tgo get -u github.com/yext/edward\n", latestVersion)
+	}
+
+}
+
+func checkUpdateAvailable(checkUpdateChan chan interface{}) {
 	updateAvailable, latestVersion, err := updates.UpdateAvailable("github.com/yext/edward", edwardVersion, filepath.Join(home.EdwardConfig.Dir, ".updatecache"), logger)
 	if err != nil {
 		fmt.Println(err)
 		logger.Println(err)
 		return
 	}
+	
+	checkUpdateChan <- updateAvailable
 	if updateAvailable {
-		fmt.Printf("A new version of Edward is available (%v), update with:\n\tgo get -u github.com/yext/edward\n", latestVersion)
+		checkUpdateChan <- latestVersion
 	}
```

Comment:
What happens if there is no update? Looks like it would wait for a message forever?
I think if you put a `defer close checkUpdateChan` at the start of this function, it will handle that case and clean up after itself nicely.